### PR TITLE
README for Shell Sort Algorithm #999

### LINKED ISSUE #999: README for Shell Sort Algorithm
Add a README.md file to explain Shell Sort Algorithm. The file should contain the following:
- Explanation
- Algorithm
- Pseudocode
- Example
- Time Complexity

Also add images/GIFs to explain if possible.

### PR DESCRIPTION
**Pull request by asha15**
Fixes #999

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Adding README.md
- 

#### Languages Used:
- 
- 

#### Files Added:
- README.md
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Shell_Sort/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+## Pseudocode
+```
+function shellSort( array : list of sortable items)
+
+length = length(array)
+gap = length / 2
+
+loop while(gap >= 1)
+  
+  for i = 0 to i <= length-1 do
```
Comment (marwahmanbir) [inline Shell_Sort/README.md]: Please fix indentation in pseudocode. 4-space, 8-space and so on...

File: Shell_Sort/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+temp = a[x]
+a[x] = a[y]
+a[y] = temp
+end function
+```
+## Time complexity
+
+Worst complexity: Depends on gap sequence<br />
+Average complexity: n*log(n)^2 or n^(3/2)<br />
```
Comment (marwahmanbir) [inline Shell_Sort/README.md]: I guess there's no need for break (``<br/>``) in above two statements.

**Comment by marwahmanbir** [comment]
Also try to squash your commits after committing the changes.

File: Shell_Sort/README.md (hunk 2 above)
Comment (asha15) [reply Shell_Sort/README.md]: @marwahmanbir I feel it's needed. otherwise both sentences are printing in the same line.

**Comment by asha15** [comment]
@marwahmanbir Done.could you review it?

**Pull request merged**